From f50c492806c77ccb297972eab8725562d904befd Mon Sep 17 00:00:00 2001
From: Robin Gong <yibin.gong@nxp.com>
Date: Thu, 20 Sep 2018 18:25:32 +0800
Subject: [PATCH 4751/5242] MLK-19734-1: sdma: add multi-fifo scripts for
 i.mx6

commit  6e5d9784281b6c58c7cf7ac31e43d252c19b0847 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add multi-fifo script for i.mx6 instead of dual fifo script, because
there is one hardware issue inside SSI ip and switch between dual fifo
and single fifo could workaroud it, so add multi-fifo scripts.

Signed-off-by: Robin Gong <yibin.gong@nxp.com>
(cherry picked from commit 85bb6546ff8cb1fc323e568d689a1e078eaca8fa)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/dma/imx-sdma.c                     |    2 +-
 include/linux/platform_data/dma-imx-sdma.h |    4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/dma/imx-sdma.c b/drivers/dma/imx-sdma.c
index 9ff2e26..2267428 100644
--- a/drivers/dma/imx-sdma.c
+++ b/drivers/dma/imx-sdma.c
@@ -1955,7 +1955,7 @@ static void sdma_issue_pending(struct dma_chan *chan)
 
 #define SDMA_SCRIPT_ADDRS_ARRAY_SIZE_V1	34
 #define SDMA_SCRIPT_ADDRS_ARRAY_SIZE_V2	38
-#define SDMA_SCRIPT_ADDRS_ARRAY_SIZE_V3	41
+#define SDMA_SCRIPT_ADDRS_ARRAY_SIZE_V3	43
 #define SDMA_SCRIPT_ADDRS_ARRAY_SIZE_V4	44
 
 static void sdma_add_scripts(struct sdma_engine *sdma,
diff --git a/include/linux/platform_data/dma-imx-sdma.h b/include/linux/platform_data/dma-imx-sdma.h
index e257209..da06645 100644
--- a/include/linux/platform_data/dma-imx-sdma.h
+++ b/include/linux/platform_data/dma-imx-sdma.h
@@ -52,10 +52,10 @@ struct sdma_script_start_addrs {
 	s32 zcanfd_2_mcu_addr;
 	s32 zqspi_2_mcu_addr;
 	s32 mcu_2_ecspi_addr;
-	/* End of v3 array */
-	s32 mcu_2_zqspi_addr;
 	s32 mcu_2_sai_addr;
 	s32 sai_2_mcu_addr;
+	/* End of v3 array */
+	s32 mcu_2_zqspi_addr;
 	/* End of v4 array */
 };
 
-- 
1.7.9.5

